### PR TITLE
cleanup refactor && adding documentation for the api

### PR DESCRIPTION
**Pull request by hanbollar**

### CODE REVIEW

File: src/component-systems/ClippingSystem.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
  */
 export class ClippingSystem extends System {
     /**
-     *
+     * ClippingSystem's default constructor
+     * // TODO - add more info
      */
     constructor() {
         super(false);
+        // TODO - how would you describe these vectors in relation to the clipping planes?
```
Comment (hanbollar) [inline src/component-systems/ClippingSystem.js:14]: @michaelthatsit - quick question for you

File: src/component-systems/ClippingSystem.js [hunk 2]
```diff
@@ -48,9 +71,11 @@ export class ClippingSystem extends System {
         });
     }
 
+// TODO - is this function still needed? i dont see it called at all?
```
Comment (hanbollar) [inline src/component-systems/ClippingSystem.js:74]: @michaelthatsit - quick question for you

File: src/component-systems/DeveloperSystem.js [hunk 3]
```diff
@@ -2,12 +2,15 @@ import * as THREE from 'three';
 import System from '../core/System';
 import { COLLIDER_CURSOR_MAP } from './RapierPhysicsSystem';
 
+// TODO - is this item still needed?
```
Comment (hanbollar) [inline src/component-systems/DeveloperSystem.js:5]: @michaelthatsit - quick question for you

File: src/component-systems/InstancingSystem.js [hunk 4]
```diff
@@ -87,6 +94,7 @@ class InstancingSystem extends System {
         entity.object3D.add(instancedMesh);
     }
 
+// TODO - can i delete the below two items?
```
Comment (hanbollar) [inline src/component-systems/InstancingSystem.js:97]: @michaelthatsit - quick question for you

File: src/utils/materialHelper.js [hunk 5]
```diff
@@ -4,6 +4,9 @@ import * as THREE from 'three';
 // mat-phong
 // "color: red; emissive: blue; specular: yellow; opacity: 0.5; shininess: 100; wireframe: true"
 
+// TODO - this should not be in this folder
+// TODO - is this even still needed? i dont see it used in the rest of the code
```
Comment (hanbollar) [inline src/utils/materialHelper.js:8]: @michaelthatsit - (the second TODO line - 8) quick question for you

**Comment by hanbollar** [comment]
@michaelthatsit - ignore the overall changes since it's still WIP

had some quick questions in this pr for when you get a chance though

File: src/component-systems/ClippingSystem.js (hunk 1 above)
Comment (michaelthatsit) [reply src/component-systems/ClippingSystem.js:14]: these are coplanar points. My understanding is they're needed to calculate the orientation and position if the plane. They're here to be reused for every plane so we're not generating a ton of vectors to be garbage collected.

File: src/component-systems/ClippingSystem.js (hunk 2 above)
Comment (michaelthatsit) [reply src/component-systems/ClippingSystem.js:74]: Yes still needed! this is called on an event whenever a new entity is added. all systems have it but not all of them utilize it.

File: src/component-systems/DeveloperSystem.js (hunk 3 above)
Comment (michaelthatsit) [reply src/component-systems/DeveloperSystem.js:5]: This whole system can be thrown out. it was for the initial demo and we will be removing this from the framework and building it on the platform.

File: src/component-systems/InstancingSystem.js (hunk 4 above)
Comment (michaelthatsit) [reply src/component-systems/InstancingSystem.js:97]: if you're referring to updated and detached component, yeah you can delete them if you're not utilizing them.

File: src/utils/materialHelper.js (hunk 5 above)
Comment (michaelthatsit) [reply src/utils/materialHelper.js:8]: lol it is not, feel free to delete it. this is a leftover experimental way to add material via data attributes

**Comment by hanbollar** [comment]
@michaelthatsit - mostly completed - there's TODO questions that are left in, i'll walk you through this PR and them tomorrow morning to confirm before i close this out

pinging you for review for a run through of descriptions separate from the TODO questions

some notes:
- there's 1 lingering build issue im tracking down from after i merged in your most recent pr
- the github action for build is currently failing due to a webpack discrepancy; i'll solve that in a followup pr since it's not a blocker locally or once a build exists

**Comment by michaelthatsit** [comment]
@hanbollar well structurally it looks great but LayoutSystem seems to have gotten lost in the merge 😅. I tried running locally but I think I'm running into that build issue you mentioned. 

We can spend tomorrow getting this across the finish line. _LFGGGGGGGG_

**Comment by hanbollar** [comment]
> @hanbollar well structurally it looks great but LayoutSystem seems to have gotten lost in the merge 😅. I tried running locally but I think I'm running into that build issue you mentioned.
> 
> We can spend tomorrow getting this across the finish line. _LFGGGGGGGG_

@michaelthatsit - woops, it was missing an additional push - those commits are up so you should see LayoutSystem now

**Comment by michaelthatsit** [comment]
looks like the src/style.css got lost too! just noticed. we'll go over it after the meeting. it runs now tho!

**Comment by hanbollar** [comment]
> looks like the src/style.css got lost too! just noticed. we'll go over it after the meeting. it runs now tho!

yea that's one of my TODO questions - i deleted that as part of the refactor 😅 - there's a few other choices i want to run by you so this meeting'll be useful - chat soon ~

**Comment by hanbollar** [comment]
@michaelthatsit - ready for re-review

some notes:
- mrjsUtils grabs the utils as a submodule for function grabbing from the namespace
- there's a slight different server visual - i'm still looking into why that's happening, lmk if you know off the top of your head and no worries if not

feel free to start running through it as an official review while i look into the above visual diff

**Comment by hanbollar** [comment]
reviewed with @michaelthatsit , pr 💣 is a go

**Pull request merged**